src/engine/input/InputManager.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Godot;
+
+public class InputManager : Node
+{
+    public override void _Ready()
+    {
+        RecalculateAttributes();
+        PauseMode = PauseModeEnum.Process;
+    }
+
+    public override void _UnhandledInput(InputEvent @event)
+    {
+        RunOnInputAttribute.AttributesWithMethods.ForEach(p => p.Item2.InputReceiver.CheckInput(@event));
```

Comment:
I cannot stop checking if something consumed the input, because e.g in the case of e_pan_left and g_move_left, they both have the same input, but in this line I don't execute any methods and therefore don't consume the input and cannot stop early. I hope you understand what I mean

Reply:
I did realize that point later. But you should still detect if *something* did something so that you can report to Godot that the input was consumed.